### PR TITLE
[improve](column) support append_data_by_selector function in const column

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/columns/column_const.h [hunk 1]
```diff
@@ -225,7 +225,7 @@ class ColumnConst final : public COWHelper<IColumn, ColumnConst> {
 
     void append_data_by_selector(MutableColumnPtr& res,
                                  const IColumn::Selector& selector) const override {
-        LOG(FATAL) << "append_data_by_selector is not supported in ColumnConst!";
+        this->template append_data_by_selector_impl<Self>(res, selector);
```
Comment (HappenLee) [inline be/src/vec/columns/column_const.h:228]: better resize the column，no need `insert_from`

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**